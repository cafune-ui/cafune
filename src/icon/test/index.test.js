import Icon from '../';
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

describe('Icon', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Icon icon="like" color="#549" size="28px" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with img', () => {
    const wrapper = shallow(<Icon icon="/someurl/img" color="#549" size="28px" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with custom prefix', () => {
    const wrapper = shallow(<Icon customPrefix="ri" tag="li" icon="/someurl/img" color="#549" size="28px" />);
    expect(wrapper).toMatchSnapshot();
  });
});
