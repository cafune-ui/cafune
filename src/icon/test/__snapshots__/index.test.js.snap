// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render properly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "color": "#549",
      "icon": "like",
      "size": "28px",
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Icon should render with custom prefix 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "color": "#549",
      "customPrefix": "ri",
      "icon": "/someurl/img",
      "size": "28px",
      "tag": "li",
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Icon should render with img 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "color": "#549",
      "icon": "/someurl/img",
      "size": "28px",
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
